src/Catel.Core/Catel.Core.Shared/Services/ObjectIdGenerator.cs

```diff
@@ -51,6 +64,78 @@ public void ReleaseIdentifier(TUniqueIdentifier identifier)
             }
         }
 
+        /// <inheritdoc />
+        public TUniqueIdentifier GetUniqueIdentifierForInstance(TObjectType instance, bool reuse = false)
+        {
+            Argument.IsNotNull("instance", instance);
+
+            var uniqueIdentifier = GetUniqueIdentifier(reuse);
```

Comment:
I don't think this is correct. If you use a WeakConditionalTable, you can actually store the object + the id. If the same instance calls the method twice, it should return the same id twice.